jsonschema-module-jackson/src/main/java/com/github/victools/jsonschema/module/jackson/KotlinJacksonModule.java

```diff
@@ -0,0 +1,62 @@
+package com.github.victools.jsonschema.module.jackson;
+
+import com.fasterxml.classmate.members.RawField;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.victools.jsonschema.generator.MemberScope;
+import java.lang.reflect.Parameter;
+import java.util.List;
+import java.util.OptionalInt;
+import kotlin.Metadata;
+
+public class KotlinJacksonModule extends JacksonModule {
+    /**
+     * Look up an alternative name for a member in the constructor parameter list.
+     * When the Kotlin compiler compiles a Kotlin data class, it creates a constructor with a parameter
+     * for each field in the data class. The parameters have generic name such as "arg0", "arg1", etc.
+     * In order to find the appropriate constructor parameter, the method first determines the index of
+     * specified member within its type member list and uses the parameter with the same index.
+     * In order to avoid erroneous overrides, this method verifies that the specified member is indeed of a
+     * Kotlin class.
+     *
+     * @param member field/method to look-up alternative property name for
+     * @return alternative property name or the base class implementation return value
+     */
+    @Override
+    protected String getPropertyNameOverrideBasedOnJsonPropertyAnnotation(MemberScope<?, ?> member) {
+        if (isKotlinType(member)) {
+            OptionalInt memberIndex = getMemberIndex(member);
+            if (memberIndex.isPresent()) {
+                Parameter[] parameters = getConstructorParameters(member);
+                Parameter parameter = parameters[memberIndex.getAsInt()];
+                JsonProperty jsonPropertyAnnotation = parameter.getAnnotation(JsonProperty.class);
+                if (jsonPropertyAnnotation != null) {
+                    String nameOverride = jsonPropertyAnnotation.value();
+                    if (isValidNameOverride(member, nameOverride)) {
+                        return nameOverride;
+                    }
+                }
+            }
+        }
+        return super.getPropertyNameOverrideBasedOnJsonPropertyAnnotation(member);
+
+    }
+
+    private OptionalInt getMemberIndex(MemberScope<?, ?> member) {
+        List<RawField> memberFields = member.getDeclaringType().getMemberFields();
+        for (int i = 0; i < memberFields.size(); i++) {
+            if (memberFields.get(i).getName().equals(member.getName())) {
+                return OptionalInt.of(i);
+            }
+        }
+        return OptionalInt.empty();
+    }
+
+    private static Parameter[] getConstructorParameters(MemberScope<?, ?> member) {
+        return member.getDeclaringType().getConstructors().get(0).getRawMember().getParameters();
```

Comment:
**suggestion (non-blocking)**: Shouldn't we assume that multiple constructors could also exist? Only considering the first might be sufficient for Kotlin data classes, but perhaps more general support of annotations on constructor parameters would be achieved by taking others into account as well?
```suggestion
        return member.getDeclaringType().getConstructors().get(0).getRawMember().getParameters();
```

Reply:
For Kotlin data classes, we can safely assume that all fields are initialized by the implicit ctor in their order in the field list. I'm afraid that might not be the case for other non-data JVM classes. 

e.g.
```java
class CtorTest{
    final int bar;
    @JsonProperty("my_bar")
    final String foo;
    public CtorTest(@JsonProperty("my_foo") int foo, String bar){
        this.bar=foo;
        this.foo=bar;
    }
}
```

How would you suggest we associate between fields and ctor parameters in the general case (non-Kotlin data classes)?

Reply:
@moranlefler I don't think we need to support such a mix of member and constructor parameter annotations.

First of all: it's probably a good idea to put this new feature behind a dedicated `JacksonOption` and then we could start with the assumption that there is one constructor with the same number of parameters as there are fields in the class and that those either match by parameter name or position.

I would have preferred matching by parameter name, but if I understood your code comment correctly, for the Kotlin data class that wouldn't work as they are generically called `arg0`, `arg1`, etc.
Therefore, we need to do it purely based on the index.
If someone else needs a different behavior in the future, we could easily extend this, but this approach should be sufficient for the Kotlin data classes as a start. No need to over-engineer it from the start.
Perhaps nobody uses those annotations on constructor parameters anymore outside of auto-generated constructors such as these. 😃